Tests/workerpool-test/WorkerPoolTest.cpp

```diff
@@ -0,0 +1,442 @@
+            TesterTask& operator=(const TesterTask&) = delete;
+
+            TesterTask(WorkerPoolTestAdministrator& parent)
+                : _parent(parent)
+            {
+                _jobMetaData.JobId = _parent.getNextJobId();
+                _jobMetaData.RepeatCnt = _parent.getRandomValue(0, 5);
+                _jobMetaData.bIsRevoked = false;
+            }
+            ~TesterTask() = default;
+
+            void Dispatch() override {
+                _parent.UpdateJobDetails(_jobMetaData.JobId);
+            }
+
+            uint32_t JobId() {
+                return _jobMetaData.JobId;
+            }
+
+            bool Revoked() {
+                return _jobMetaData.bIsRevoked;
+            }
+
+            void Revoked(bool bRevoked) {
+                _jobMetaData.bIsRevoked = bRevoked;
+            }
+
+            uint32_t RepeatCount() {
+                return _jobMetaData.RepeatCnt;
+            }
+
+            void RepeatCount(uint32_t repeatCnt) {
+                _jobMetaData.RepeatCnt = repeatCnt;
+            }
+        private:
+            WorkerPoolTestAdministrator& _parent;
+            WorkerPoolTestAdministrator::TestJobMetaData _jobMetaData;
+
+        };
+
+        class MonitorTask : public Core::IDispatch
+        {
+        public:
+            MonitorTask(const MonitorTask&) = delete;
+            MonitorTask& operator=(const MonitorTask&) = delete;
+
+            MonitorTask(WorkerPoolTestAdministrator& parent)
+                : _parent(parent) {
+
+                }
+
+            void Dispatch() override {
+                _parent.AnalyseTestTask();
+            }
+
+        private:
+            WorkerPoolTestAdministrator& _parent;
+
+        };
+
+    private:
+        class WorkerPoolImplementation : public Core::WorkerPool {
+
+        private:
+            class Dispatcher : public Core::ThreadPool::IDispatcher {
+            public:
+                Dispatcher(const Dispatcher&) = delete;
+                Dispatcher& operator=(const Dispatcher&) = delete;
+
+                Dispatcher() = default;
+                ~Dispatcher() override = default;
+
+            private:
+                void Initialize() override { }
+                void Deinitialize() override { }
+                void Dispatch(Core::IDispatch* job) override {
+                    job->Dispatch();
+                }
+
+            };
+
+        public:
+            WorkerPoolImplementation() = delete;
+            WorkerPoolImplementation(const WorkerPoolImplementation&) = delete;
+            WorkerPoolImplementation& operator=(const WorkerPoolImplementation&) = delete;
+
+            WorkerPoolImplementation(const uint8_t threads, const uint32_t stackSize, const uint32_t queueSize)
+                : WorkerPool(threads, stackSize, queueSize, &_dispatcher)
+                , _dispatcher(){ }
+
+            ~WorkerPoolImplementation()
+            {
+                printf("%s[ %d ] Called \n", __FUNCTION__, __LINE__);
```

Comment:
Are these kind of printfs useful for the user of the testtool or were these for helping in development. In case of the latter perhaps better to remove them?

Reply:
Removed all the printf added for debugging purpose.